### DISCUSSION TITLE
Edge Waypoint can not be grabed in Firefox Browser

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#870] [Q&A]
I currently have a strange effect with my modeller project. When running the modeler in Theia using Firefox Browser I can not grab a Edge Waypoint and move it. 

![Peek 2023-01-20 14-33](https://user-images.githubusercontent.com/1339124/213707765-be3397f1-14b8-43d4-8d93-5171ba7e0f8c.gif)

But every thing else works normal. Using chrome browser all works fine.

You can see this effect in the online demo of coffee-editor too:  https://try.theia-cloud.io/?appDef=coffee-editor

If you run the demo in Firefox and select a waypoint, you can not move it on the first click. You need to click the waypoint twice to get it moveable. 
But when using Chrome the waypoint is always moveable. 

Maybe I have implemented a `selectionService` in a wrong way causing this issue. But in my case I can not move a waypoint in Firefox even if I click several times on the waypoint.

Does anybody know about this kind of issue?

**Comment by martin-fleck-at** [comment]
@rsoika Good catch! I actually can reproduce this on the workflow example as well. However, from a first glance I did not see anything obvious missing. I will do a deeper dive tomorrow and let you know if I find the issue. Thank you for the report!

**Reply by rsoika** [reply]
...maybe this is somehow related to the fix of [#657](https://github.com/eclipse-glsp/glsp/issues/657) ? But now it does not really depend on the count of routing points....

**Reply by martin-fleck-at** [reply]
I created https://github.com/eclipse-glsp/glsp/issues/877 to track the issue and already proposed a fix in https://github.com/eclipse-glsp/glsp-client/pull/224. It seems to me like Firefox is sending an additional `mouseEnter` event after the mouse down which caused us to reset the drag position. However, I don't think we need to reset the position at all on `mouseEnter`.
